packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -625,6 +625,12 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
   /// Animates the position such that the given object is as visible as possible
   /// by just scrolling this position.
   ///
+  /// The optional `targetRenderObject` parameter is used to determine which area
+  /// of that object should be as visible as possible. If `targetRenderObject` is null,
+  /// the entire [RenderObject] (as defined by its [RenderObject.paintBounds])
+  /// will be as visible as possible. If `targetRenderObject` is provided it should be
+  /// a descendant of the object.
+  ///
```

Comment:
Why wouldn't you just specify the `targetRenderObject` as the `object` parameter?

Reply:
@dnfield This is because `viewport.getOffsetToReveal` cannot work well traverse multiple viewports, also see https://github.com/flutter/flutter/blob/38834d36d3320fce0f43ef7df29846b0d073fd6f/packages/flutter/lib/src/rendering/viewport.dart#L68-L70.

Reply:
Why can't we just traverse up the viewports and get the offset for all of them?

Reply:
@dnfield 
Yes, we are traverse up `viewport` by the superior calling,
https://github.com/flutter/flutter/blob/78fdd52a0fcf9a75686079ad72b46c938bec4c87/packages/flutter/lib/src/widgets/scrollable.dart#L320-L328

Here just a correction to the area to be displayed.

Reply:
Ohhhh I see now, we're using this parameter to memoize which object we were initially trying to center and let the viewports know that's the one - we don't expect callers to actually pass this in as a duplicate of object. Is that correct?

Reply:
@dnfield 
You are absolutely correct.
Actually, if the caller does this, it has the same effect as setting it to null, which means the target area is itself.